compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -599,7 +599,16 @@ fn link_dwarf_object<'a>(
     cg_results: &CodegenResults,
     executable_out_filename: &Path,
 ) {
-    let dwp_out_filename = executable_out_filename.with_extension("dwp");
+    let dwp_extension = executable_out_filename
+        .extension()
+        .map(|previous_extension| {
+            let mut previous_extension = previous_extension.to_os_string();
+            previous_extension.push(".dwp");
+            previous_extension
+        })
+        .unwrap_or_else(|| "dwp".into());
+    let mut dwp_out_filename = executable_out_filename.to_path_buf();
+    dwp_out_filename.set_extension(dwp_extension);
```

Comment:
Is just this just always adding ".dwp"? If so, this might be easier to read:

```suggestion
    let mut dwp_out_filename = executable_out_filename.to_path_buf().into_os_string();
    dwp_out_filename.push(".dwp");
```

Reply:
Yeah that's a lot neater, thanks.